### DISCUSSION TITLE
Search and replace all occurrences (in open document)

### DISCUSSION THREAD

**Discussion by idealemu** [vkbo/novelWriter#2032] [Q&A] [answered]
Hi,

As far as I can see, the Search and Replace menu does not have an option to replace all found occurrences, but rather needs to go through one by one ? 
The manual does not mention anything about this (version 2.5.2), is there currently a way to do this ?

**Comment by idealemu** [comment]
(I am doing this specifically in the case of replacing quotes after copying in a bunch of text, as the dialogue highlighting feature didn't seem to recognize the pasted quotes, I do not have auto-replace enabled, but I also don't think it can retroactively replace existing text ?)

**Reply by vkbo** [reply]
No, the auto-replace is only triggered on typing. However, your problem is a common enough problem, so there are two entries at the bottom of the format menu to replace straight quotes in selected text with the quote symbols you've selected in Preferences.

The dialogue highlighting will *only* highlight the quote symbols selected in Preferences, which default to the English symbols. If you do wish to use standard keyboard straight quotes, you can change this in Preferences.

**Comment by vkbo** [comment, marked as answer]
There is no replace all feature, no. You can do it quite rapidly by using the keyboard shortcut though, and I've done it in very large document with no issue, or multiple documents if you turn on the switch to keep searching in next file, so I've never seen a big reason for adding it.

The keyboard shortcut is `Ctrl+Shift+1`, and is mentioned in the menu.

**Reply by wessleyn** [reply]
Does this mean they'll be no such feature in the future and we to adapt or? Cause I can really wait

**Reply by vkbo** [reply]
See #545 

The main issue is that the search feature is provided by the underlying Qt library, which does not include a find all feature, so there isn't a safe way to preview a replace all. The risk of messing up the document is therefore there. Looping through the search and replace yourself is safer because you'll get that preview.

But there is a feature request for just implementing this as a loop, to be used at your own risk.

**Reply by wessleyn** [reply]
Thanks, surely didn't consider all sides of the coin, surely no one know the app better than the developer
